packages/toolkit/src/createSlice.ts

```diff
@@ -460,33 +742,18 @@ type SliceActionType<
  * @public
  */
 export type CaseReducerActions<
-  CaseReducers extends SliceCaseReducers<any>,
+  CaseReducers extends CreatorCaseReducers<State>,
   SliceName extends string,
-> = {
-  [Type in keyof CaseReducers]: CaseReducers[Type] extends infer Definition
```

Comment:
you [can](https://github.com/reduxjs/redux-toolkit/pull/3894/commits/5f5fe7952630acc7b28de4d0e371522a23922cba), but you lose the ergonomics of `ReducerNamesOfType`. You end up always having to do
```ts
{
  [ReducerName in keyof CaseReducers]: CaseReducers[ReducerName] extends ReducerDefinition<typeof myCustomType> ? MyActualLogicHere : never
}
```

Definitions that aren't recognised by any handlers also get mapped to `unknown` instead of not being exposed at all: 
```ts
const slice = createSlice({
  name,
  initialState,
  reducers: {
    foo: "banana!"
  }
})

slice.actions.foo // previously not exposed as a key, now is `unknown`
```

the types are also having to do more work because there's 1) more intersections and 2) we're converting any `never`s into `unknown` ("foo" & never = never, "foo" & unknown = "foo")

Reply:
removed `ReducerNamesOfType` and updated docs with conditional needed instead